cloud/wfm/src/process-instance/index.ts

```diff
@@ -0,0 +1,78 @@
+import * as Promise from 'bluebird';
+import { EventEmitter } from 'eventemitter3';
+import * as _ from 'lodash';
+import { Process } from '../process';
+import { Task } from '../task';
+
+export interface InstanceStepEventData<T extends ProcessInstance>  {
+  instance: T;
+  task: Task;
+}
+
+export interface InstanceEventData<T extends ProcessInstance> {
+  instance: T;
+}
+
+/**
+ * The executable instance of a Process
+ */
+export interface ProcessInstance extends EventEmitter {
+  assigneeId: string;
+
+  getTasks(): Promise<Task[]>;
+  getCurrentTask(): Promise<Task>;
+  /**
+   * @return A promise that resolves to the next task in it's list, can be async
+   */
+  next(): Promise<Task>;
+
+  getId(): string;
+
+  /**
+   * Event emitted when
+   */
+  on(event: 'task:change', handler: (e: InstanceStepEventData<this>) => any): this;
+  on(event: 'process:done', handler: (e: InstanceEventData<this>) => any): this;
+}
+
+class InstanceImpl extends EventEmitter implements ProcessInstance {
```

Comment:
It's this guy: https://github.com/primus/eventemitter3, runs everywhere and is compatible with nodejs' api